pkg/query-service/app/integrations/thirdPartyApi/translator.go

```diff
@@ -1,598 +1,514 @@
 package thirdPartyApi
 
 import (
+	"fmt"
 	"net"
+	"time"
 
-	v3 "github.com/SigNoz/signoz/pkg/query-service/model/v3"
+	qbtypes "github.com/SigNoz/signoz/pkg/types/querybuildertypes/querybuildertypesv5"
+	"github.com/SigNoz/signoz/pkg/types/telemetrytypes"
 )
 
 const (
-	urlPathKey    = "http.url"
-	serverNameKey = "net.peer.name"
+	urlPathKeyLegacy    = "http.url"
+	serverNameKeyLegacy = "net.peer.name"
+
+	urlPathKeyCurrent    = "url.full"
+	serverNameKeyCurrent = "server.address"
 )
 
-var defaultStepInterval int64 = 60
+var defaultStepInterval = 60 * time.Second
+
+// SemconvFieldMapping defines the mapping between legacy and current semconv
+type SemconvFieldMapping struct {
+	LegacyField  string
+	CurrentField string
+	FieldType    telemetrytypes.FieldDataType
+	Context      telemetrytypes.FieldContext
+}
+
+var semconvMappings = []SemconvFieldMapping{
+	{
+		LegacyField:  urlPathKeyLegacy,
+		CurrentField: urlPathKeyCurrent,
+		FieldType:    telemetrytypes.FieldDataTypeString,
+		Context:      telemetrytypes.FieldContextAttribute,
+	},
+	{
+		LegacyField:  serverNameKeyLegacy,
+		CurrentField: serverNameKeyCurrent,
+		FieldType:    telemetrytypes.FieldDataTypeString,
+		Context:      telemetrytypes.FieldContextAttribute,
+	},
+}
+
+// CreateDualSemconvGroupByKeys creates group by keys supporting both semconv versions
+func CreateDualSemconvGroupByKeysServer() []qbtypes.GroupByKey {
+	return createDualSemconvGroupBy(semconvMappings[1]) // server name mapping
+}
+
+func CreateDualSemconvGroupByKeysUrl() []qbtypes.GroupByKey {
+	return createDualSemconvGroupBy(semconvMappings[0]) // url path mapping
+}
+
+func createDualSemconvGroupBy(mapping SemconvFieldMapping) []qbtypes.GroupByKey {
+	return []qbtypes.GroupByKey{
+		{
+			TelemetryFieldKey: telemetrytypes.TelemetryFieldKey{
+				Name:          mapping.CurrentField,
+				FieldDataType: mapping.FieldType,
+				FieldContext:  mapping.Context,
+				Signal:        telemetrytypes.SignalTraces,
+			},
+		},
+		{
+			TelemetryFieldKey: telemetrytypes.TelemetryFieldKey{
+				Name:          mapping.LegacyField,
+				FieldDataType: mapping.FieldType,
+				FieldContext:  mapping.Context,
+				Signal:        telemetrytypes.SignalTraces,
+			},
+		},
+	}
+}
+
+func MergeSemconvColumns(result *qbtypes.QueryRangeResponse) *qbtypes.QueryRangeResponse {
+	if result == nil || result.Data.Results == nil {
+		return result
+	}
+
+	for _, res := range result.Data.Results {
+		scalarData, ok := res.(*qbtypes.ScalarData)
+		if !ok {
+			continue
+		}
+
+		serverAddrIdx := -1
+		netPeerIdx := -1
+
+		for i, col := range scalarData.Columns {
+			if col.Name == serverNameKeyCurrent {
+				serverAddrIdx = i
+			} else if col.Name == serverNameKeyLegacy {
+				netPeerIdx = i
```

Comment:
this doesn't seem right

Reply:
please add test cases specifc to this function so that I can get proper idea.

Reply:
Please use same naming convention to avoid confusion. Above you are using `legacy` and `current` for serverName, please use the same here as well.